autopush-common/src/reliability.rs

```diff
@@ -66,35 +68,29 @@ impl PushReliability {
         if reliability_dsn.is_none() {
             debug!("🔍 No reliability DSN declared.");
             return Ok(Self {
-                client: None,
+                pool: None,
                 db: db.clone(),
             });
         };
 
-        let client = if let Some(dsn) = reliability_dsn {
-            let redis_client = redis::Client::open(dsn.clone()).map_err(|e| {
-                ApcErrorKind::GeneralError(format!("Could not connect to redis server: {:?}", e))
-            })?;
-            //Verify that we can connect to the server.
-            match redis_client.get_connection_with_timeout(CONNECTION_EXPIRATION) {
-                Ok(_) => {
-                    info!("🔍 Connected to reliability server at {dsn}");
-                    Some(Arc::new(redis_client))
-                }
-                Err(e) => {
-                    // This is a "soft" error. It does impact whether or not reliability will work but it's
-                    // not something that should prevent the server from running.
-                    error!("🔍 Invalid reliability DNS specified: {e} for DSN: {dsn}.");
-                    None
-                }
-            }
-        } else {
-            trace!("🔍 Empty Redis DSN?");
-            None
+        let pool = {
+            let config: deadpool_redis::Config = Config::from_url(reliability_dsn.clone().unwrap());
```

Comment:
You can kill this unwrap by switching the first line of this method to:

```rust
let Some(reliability_dsn) = reliability_dsn else {
```

```suggestion
            let config: deadpool_redis::Config = Config::from_url(&reliability_dsn);
```